app/models/metadatum.rb

```diff
@@ -277,7 +277,7 @@ def self.new_without_save(sample, key, value)
   def validated_value
     # Special case for Location objects
     if metadata_field.base_type == Metadatum::LOCATION_TYPE
-      location_id ? Location.find(location_id).attributes : string_validated_value
+      Hash[Location::DEFAULT_LOCATION_FIELDS.map { |k| [k, location[k]] }]
```

Comment:
I think you still want the conditional around this for `location ? : string_validated_value`. If location_id is nil then there's no Location object and it's just string_validated_value.